### PR TITLE
Quality of life improvements

### PR DESCRIPTION
**Pull request by droogie**
Two new quality of life improvements, one which allows you to enable a topdown format of the calltree which feels significantly more natural (set as default). The other is a lock button which will stop the calltree from being updated while navigating

screen_shot_2022-03-25_23:01-51-577466558

### CODE REVIEW

**Comment by elbiazo** [comment]
Hey Droogie,

Thank you for your contribution 😊. I will review the code and merge it in by tomorrow!

**Comment by droogie** [comment]
Sounds great! There are definitely some concerns I have with the approach I took. I thought it would be simplest to just reverse the tree after your loop built it, the issue here is that the work is essentially being done twice. The binaries I've tested were quite small so this was not an issue, but the fact that you have opened #7, I imagine there might be some timing delays or other concerns with larger binaries. 

The second was that I thought it would be nice to reflect name changes in the call tree immediately. In default usage this is not a big deal, but with the lock feature if you're navigating through a bunch of the references and renaming them it would be nice to immediately see the change. It sounds like it wouldn't be too hard to add but due to my limited experience with QT and trying to avoid introducing a bunch of messy spaghetti code to your project, I haven't put much thought into an approach. Maybe once caching is implemented you can just update the local cache and reflect the change to the tree regardless of lock state, but not sure if there's a callback that will trigger on rename or if it would require polling

**Comment by elbiazo** [comment]
I like the top-down view. It is reporting function itself though. 
image

For the lock feature, I have been thinking about how I would want to do this. At first I wanted to do something like how binja does this with cross reference. However, instead of creating another whole dock,  i was thinking making multi-tab feature so you can "lock it". I would probably add this in for now but will change it later. Maybe something like this
https://doc.qt.io/qt-5/qtabwidget.html#details

Another thing is that when you unlock, it does not refresh to current view.

TLDR:
I would like to fix these two before merging this in.
1. Function also counts itself on top-down view.
2. When you unlock, it does not refresh to current view.

**Comment by droogie** [comment]
I actually manually added it to display itself in the call window, I just personally liked it being there but it can be removed by getting rid of the conditional here: https://github.com/elbiazo/calltree/pull/8/commits/adcbb9ae1a1b29ba94ec2973f52bb7e175911505#diff-d1f389034f3b4b96b4d8544628d78d56fa02da0891fbe50f27574e14b3447053R200

Good call on refreshing to current view the moment it unlocks

**Comment by droogie** [comment]
OK I've fixed those concerns. The settings options are now a drop down between the formats "Top Down (Display Current Function), Top Down and Bottom Up"

I also fixed the refresh upon unlocking. I didn't know the best way to approach the problem so I just created a `get_bv()` function in `CallTreeLayout` so I could obtain the current `bv.offset`. 

screen_shot_2022-03-28_19:15-16-144488172

**Comment by elbiazo** [comment]
I should have done more testing before merging. It seems like now the index incrementing and decrementing does not work. I can definitely checkout the code and fix it maybe by end of the week.  Unless you have some time to do so but no worries :)

**Comment by droogie** [comment]
interesting... I'll take a look, I tested that a few days ago but maybe some recent changes broke it

**Comment by elbiazo** [comment]
No worries. I'll also try to look throughout this week.
